### PR TITLE
Recording sim: Moved global variables into the board recording state

### PR DESCRIPTION
**Pull request by srietkerk**
We don't want to cause memory leaks if the board gets destroyed. By moving the global variables into the recording state object that gets initialized in `dalboard.ts`, the variables should get destroyed if the board gets reconstructed. 

Done in thanks to Joey's comment here: https://github.com/microsoft/pxt-microbit/pull/5082#discussion_r1178571516

Work related to https://github.com/microsoft/pxt-microbit/pull/5082

### CODE REVIEW

File: sim/state/record-audio.ts [hunk 1]
```diff
@@ -73,24 +77,30 @@ namespace pxsim.record {
...
-        return recorder ? recorder.state == "recording" : false;
+        const b = board();
+        if (!b) return false;
+        return b.recordingState.recorder ? b.recordingState.recorder.state == "recording" : false;
     }
 
     export function audioIsStopped(): boolean {
-        return recorder ? !audioPlaying && recorder.state == "inactive" : true;
+        const b = board();
+        if (!b) return false;
```
Comment (jwunderl) [inline sim/state/record-audio.ts:102]: Should this be true? I'm assuming stopped is basically just "is not playing"? Might be wrong on that assumption

**Review by jwunderl — APPROVED** [review]

File: sim/state/record-audio.ts (hunk 1 above)
Comment (srietkerk) [reply sim/state/record-audio.ts:102]: Ah, yes, this should be true. Thank you!

**Review by thsparks — APPROVED** [review]

File: sim/state/record-audio.ts [hunk 2]
```diff
@@ -1,42 +1,42 @@
...
-    let recorder: MediaRecorder;
-    let chunks: Blob[];
-    let audioURL: string;
-    let recording: HTMLAudioElement;
-    let audioPlaying: boolean = false;
-
     export async function record(): Promise<void> {
         //request permission is asynchronous
         let b = board();
+        let recordingState = b.recordingState;
```
Comment (kimprice) [inline sim/state/record-audio.ts:16]: nit: do you want to scope down `recordingState`? You only use it within the `try` so you could wait to define it there.

**Review by kimprice — APPROVED** [review]
Looks good. Just a suggestion on scoping down `recordingState`.

**Pull request merged**